perceval/backends/core/github.py

```diff
@@ -667,6 +680,8 @@ def fetch(self, url, payload=None, headers=None, method=HttpClient.GET, stream=F
         response = super().fetch(url, payload, headers, method, stream, verify)
 
         if not self.from_archive:
+            if self._need_check_tokens():
```

Comment:
This is my main concern with the patch and there's a bug in here. When `self.update_rate_limit` is called, it's updating the `rate_limit` to the rate limit obtained in the last request (see line 680). Therefore, when `sleep_for_rate_limit` is called the next time it will be checking the rate of the last token and not the new one selected.

If someone sets a big value in `min_rate_to_sleep`, the code will stop even when there are tokens with enough rate.

Reply:
I will take a look at this, but I see your point.